### ISSUE TITLE
kvm broken under d2g

### ISSUE THREAD

**Issue by petemoore** [taskcluster/taskcluster#7082]
See https://github.com/MozillaSecurity/fuzzing-tc-config/pull/171#issuecomment-2155776248

**Comment by petemoore** [comment]
So it turns out kvm isn't broken per se, but the podman run command that d2g generates is not equivalent to the original docker invocation that docker-worker used to make, due to differences between default behaviour of docker and podman. In this particular case, code inside the container queries group ownership of `/dev/kvm` and ensures that the current user (`worker`) is a member of the group. Inside the podman container, `/dev/kvm` appears to be owned by group `nogroup` since the group is not mapped to a group on the host. Under docker, the group inside the container does map to the host group `kvm` and so the user in the container is able to access the kvm device.

There are several alternative solutions here:

1. For the task duration of a task that has the kvm feature enabled, make `/dev/kvm` read/writable for all users. This doesn't play well when Generic Worker will later support running multiple workers on the same machine in parallel.
2. Potentially modify /etc/subgid and use e.g. newgidmap to map groups inside the podman container. Probably do this by generating a command to do it in d2g, rather than inside Generic Worker, because Generic Worker shouldn't need to know about this.
3. Use docker instead of podman when Docker Worker tasks have privileged capability enabled. This wouldn't be as secure as using podman.
4. Allow task to specify whether to use docker/podman in Docker Worker task payload. We don't really want to keep supporting Docker Worker payloads, d2g was intended to be an interim measure, so not ideal.
5. Migrate the existing tasks to Generic Worker format (avoid d2g) and use docker. This solves all the issues except that podman may be more secure than docker, and also there could be further similar issues that get thrown up in FXCI like this, so it only solves the issue for the current use case.
6. Like 5, but use podman and newgidmap in the new task definitions rather than docker. If we get that working though, we might as well make it part of d2g (option 2).

**Comment by petemoore** [comment]
I think 3 is my favourite _first_ choice because it doesn't initially appear to have any obvious drawbacks.

**Comment by petemoore** [comment]
I've gone with option 4 for now, as it is quite a trivial fix and unblocks the issue. We can consider switching to an alternative solution at some point in the future.
